tests/test_primod/test_preprocessing_ribamod.py

```diff
@@ -250,14 +250,19 @@ def test_nullify_on_write(
         ribasim_dll_dependency="c",
     )
     df = coupled_models.ribasim_model.basin.static.df
-    # Basin 2 is coupled, so drainage & infiltration should be NaN.
-    assert (
-        df.loc[df["node_id"] == 2, ["drainage", "infiltration"]].isna().all(axis=None)
-    )
-    # Basin 3 is uncoupled, terms should be preserved.
-    assert (
-        df.loc[df["node_id"] == 3, ["drainage", "infiltration"]].notna().all(axis=None)
-    )
+    if df is not None:
+        # Basin 2 is coupled, so drainage & infiltration should be NaN.
+        assert (
+            df.loc[df["node_id"] == 2, ["drainage", "infiltration"]]
+            .isna()
+            .all(axis=None)
+        )
+        # Basin 3 is uncoupled, terms should be preserved.
+        assert (
+            df.loc[df["node_id"] == 3, ["drainage", "infiltration"]]
+            .notna()
+            .all(axis=None)
+        )
```

Comment:
I think this part should be removed.
Agreed @Huite?

Reply:
I would suggest setting a value for the dataframe instead in the test.

I think it would be fine to test it twice: once with the model as is (which shouldn't raise errors) and a second time, after which you set the dataframe with e.g. zero values for all columns. Then do these asserts.

Reply:
Done